libdevcore/RLP.h

```diff
@@ -63,415 +86,780 @@ template <class T> struct Converter { static T convert(RLP const&, int) { BOOST_
-		{
-			if (_flags & ThrowOnFail)
-				BOOST_THROW_EXCEPTION(BadCast());
-			else
-				return 0;
-		}
-
-		auto p = payload();
-		if (p.size() > intTraits<_T>::maxSize && (_flags & FailIfTooBig))
-		{
-			if (_flags & ThrowOnFail)
-				BOOST_THROW_EXCEPTION(BadCast());
-			else
-				return 0;
-		}
-
-		return fromBigEndian<_T>(p);
-	}
-
-	template <class _N> _N toHash(int _flags = Strict) const
-	{
-		requireGood();
-		auto p = payload();
-		auto l = p.size();
-		if (!isData() || (l > _N::size && (_flags & FailIfTooBig)) || (l < _N::size && (_flags & FailIfTooSmall)))
-		{
-			if (_flags & ThrowOnFail)
-				BOOST_THROW_EXCEPTION(BadCast());
-			else
-				return _N();
-		}
-
-		_N ret;
-		size_t s = std::min<size_t>(_N::size, l);
-		memcpy(ret.data() + _N::size - s, p.data(), s);
-		return ret;
-	}
-
-	/// Converts to RLPs collection object. Useful if you need random access to sub items or will iterate over multiple times.
-	RLPs toList(int _flags = Strict) const;
-
-	/// @returns the data payload. Valid for all types.
-	bytesConstRef payload() const { auto l = length(); if (l > m_data.size()) BOOST_THROW_EXCEPTION(BadRLP()); return m_data.cropped(payloadOffset(), l); }
-
-	/// @returns the theoretical size of this item as encoded in the data.
-	/// @note Under normal circumstances, is equivalent to m_data.size() - use that unless you know it won't work.
-	size_t actualSize() const;
+    /// Conversion flags
+    enum
+    {
+        AllowNonCanon = 1,
+        ThrowOnFail = 4,
+        FailIfTooBig = 8,
+        FailIfTooSmall = 16,
+        Strict = ThrowOnFail | FailIfTooBig,
+        VeryStrict = ThrowOnFail | FailIfTooBig | FailIfTooSmall,
+        LaissezFaire = AllowNonCanon
+    };
+
+    using Strictness = int;
+
+    /// Construct a null node.
+    RLP() {}
+
+    /// Construct a node of value given in the bytes.
+    explicit RLP(bytesConstRef _d, Strictness _s = VeryStrict);
+
+    /// Construct a node of value given in the bytes.
+    explicit RLP(bytes const& _d, Strictness _s = VeryStrict) : RLP(&_d, _s) {}
+
+    /// Construct a node to read RLP data in the bytes given.
+    RLP(byte const* _b, unsigned _s, Strictness _st = VeryStrict)
+      : RLP(bytesConstRef(_b, _s), _st)
+    {}
+
+    /// Construct a node to read RLP data in the string.
+    explicit RLP(std::string const& _s, Strictness _st = VeryStrict)
+      : RLP(bytesConstRef((byte const*)_s.data(), _s.size()), _st)
+    {}
+
+    /// The bare data of the RLP.
+    bytesConstRef data() const { return m_data; }
+
+    /// @returns true if the RLP is non-null.
+    explicit operator bool() const { return !isNull(); }
+
+    /// No value.
+    bool isNull() const { return m_data.size() == 0; }
+
+    /// Contains a zero-length string or zero-length list.
+    bool isEmpty() const
+    {
+        return !isNull() && (m_data[0] == c_rlpDataImmLenStart ||
+                                m_data[0] == c_rlpListStart);
```

Comment:
I don't understand this alignment. What did it align with?

Reply:
four more indentations than `(`.

Reply:
Weird, but I think it's continuation (4) from `(`.